### DISCUSSION TITLE
OpenAI integration support for realtime api

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#3765] [Ideas]
### Describe the feature or potential improvement

Tracing integration for realtime api: https://platform.openai.com/docs/guides/realtime

**Comment by kubephil** [comment]
Just quickly wanted to drop features that would be cool:
- Overview on the pricing split by transcription, token and audio (really usnsure how I can get that out of the API as of right now -> maybe "postprocess" the transcripts to get the tokens from it and timing on the audio files ?)
- the ability to upload the audio files or somehow link the location of them. Let's say I upload them to s3 so that I could link the bucket location there later
- trace my tool calls that the realtime API did (this I could see how I can implement this as I can extract them fro the openai events already )
I'm open to ideas if there are already ways to make it work in langfuse or support on the feature development once we have a rough plan on how it would work

**Reply by marcklingen** [reply]
cc @hassiebp as this is related to planned multi-modal and cost tracking changes

**Comment by ghost** [comment]
I'd be very interested in getting some "basic" support for use with the [Realtime API beta](https://github.com/openai/openai-realtime-api-beta).  
Just being able to see the generations in Langfuse with an overview of the total price would be a big step forward.

**Comment by yuyuma** [comment]
Very interested in Langfuse support for realtime here.

I just want to add that I requested and the LiteLLM team quickly turned around basic support for realtime proxying (see: https://docs.litellm.ai/docs/realtime), including basic observability via their Langfuse integration.  I would love to see Langfuse properly parse (a filtered set of) the realtime server event data stream so as to be able to display them nicely within the Langfuse console, as well as extract the usage details / associated costs.

I can provide actual traces in Langfuse cloud in case someone from the team needs a reference while working on this.

**Comment by marcklingen** [comment]
I've just create a new thread to capture notes on requirements (vendor-agnostic) in order to keep this thread more focused on the openai integration.

https://github.com/orgs/langfuse/discussions/4757

**Comment by psr-ai** [comment]
Is this currently being worked upon?
